mlrun/projects/project.py

```diff
@@ -3159,6 +3169,9 @@ def build_image(
                 force_build=True,
             )
 
+            # Get the enriched target dir from the function
+            self.spec.build.source_target_dir = function.spec.build.source_target_dir
```

Comment:
Don't we want to allow passing `source_target_dir` to `build_image` as parameter? Also, I'm not sure - does it matter if we pull at runtime or not? If we pull at runtime then this parameter has no meaning, isn't it so?

Reply:
About passing `source_code_target_dir` as a parameter we use `target_dir` for it but it can be relative and then we need the full path which we get from the build function response.
About target dir with pull at runtime is should work and pull to the source code target dir. I added the enrichment in `mlrun.projects.pipelines.enrich_function_object` good catch 👍